packages/shared_preferences/shared_preferences/example/integration_test/shared_preferences_test.dart

```diff
@@ -97,5 +89,48 @@ void main() {
       // The last write should win.
       expect(preferences.getInt('int'), writeCount);
     });
+  }
+
+  group('SharedPreferences', () {
+    setUp(() async {
+      preferences = await SharedPreferences.getInstance();
+    });
+
+    tearDown(() {
+      preferences.clear();
+      SharedPreferences.resetStatic();
+    });
+
+    runAllTests();
+  });
+
+  group('setPrefix', () {
+    setUp(() async {
+      SharedPreferences.resetStatic();
+      SharedPreferences.setPrefix('prefix.');
+      preferences = await SharedPreferences.getInstance();
+    });
+
+    tearDown(() {
+      preferences.clear();
+      SharedPreferences.resetStatic();
+    });
+
+    runAllTests();
+  });
+
+  group('setNoPrefix', () {
+    setUp(() async {
+      SharedPreferences.resetStatic();
+      SharedPreferences.setPrefix('');
+      preferences = await SharedPreferences.getInstance();
+    });
+
+    tearDown(() {
+      preferences.clear();
+      SharedPreferences.resetStatic();
+    });
+
+    runAllTests();
   });
```

Comment:
There should be tests of the `UnimplementedError` rewriting as well.

Reply:
Wouldn't that involve importing the platform, which is not allowed? (I think)

Reply:
You can just make a completely empty implementation that `extends` the base platform interface class (and thus inherts its unimplemented behavior), and test with the platform instance set to that.

Oh, I left this comment in the integration test file though; I meant to leave them in the Dart unit test file since it's easily testable there.